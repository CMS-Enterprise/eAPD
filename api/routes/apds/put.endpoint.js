const {
  getDB,
  getFullPath,
  login,
  request,
  unauthenticatedTest,
  unauthorizedTest
} = require('../../utils.endpoint');

describe('APD endpoint | PUT /apds/:id', async () => {
  const db = getDB();
  beforeAll(() => db.seed.run());
  afterAll(() => db.destroy());

  const url = apdID => getFullPath(`/apds/${apdID}`);

  unauthenticatedTest('put', url(1));
  unauthorizedTest('put', url(1));

  describe('when authenticated as a user with permission', async () => {
    let cookies;
    beforeAll(async () => {
      cookies = await login();
    });

    it('with a non-existant apd ID', async () => {
      const { response, body } = await request.put(url(9000), {
        jar: cookies,
        json: true
      });

      expect(response.statusCode).toEqual(404);
      expect(body).toMatchSnapshot();
    });

    it(`with an APD in a state other than the user's state`, async () => {
      const { response, body } = await request.put(url(4001), {
        jar: cookies,
        json: true
      });

      expect(response.statusCode).toEqual(404);
      expect(body).toMatchSnapshot();
    });

    it('with a valid update', async () => {
      const { response, body } = await request.put(url(4000), {
        jar: cookies,
        json: {
          programOverview: 'new overview'
        }
      });

      expect(response.statusCode).toEqual(200);
      expect(body).toMatchSnapshot();
    });
  });
});
