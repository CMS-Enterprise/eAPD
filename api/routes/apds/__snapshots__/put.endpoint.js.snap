// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`APD endpoint | PUT /apds/:id when authenticated as a user with permission with a non-existant apd ID 1`] = `undefined`;

exports[`APD endpoint | PUT /apds/:id when authenticated as a user with permission with a valid update 1`] = `
Object {
  "activities": Array [
    Object {
      "alternatives": null,
      "contractorResources": Array [
        Object {
          "description": null,
          "end": null,
          "files": Array [
            Object {
              "id": 5002,
              "size": null,
            },
          ],
          "hourlyData": Array [],
          "id": 4300,
          "name": null,
          "start": null,
          "useHourly": false,
          "years": Array [],
        },
      ],
      "costAllocation": Array [],
      "costAllocationNarrative": Object {
        "methodology": null,
        "otherSources": null,
      },
      "description": "Some text goes here",
      "expenses": Array [],
      "files": Array [
        Object {
          "id": 5000,
          "in": "here",
          "size": null,
          "some": "metadata",
        },
        Object {
          "id": 5004,
          "size": null,
        },
      ],
      "fundingSource": null,
      "goals": Array [
        Object {
          "description": "Be a super successful artist",
          "id": 4200,
          "objective": "Paint a pretty picture",
        },
        Object {
          "description": "Win a Nobel prize for physics",
          "id": 4201,
          "objective": "Discover a new particle",
        },
        Object {
          "description": "Go on Ellen",
          "id": 4202,
          "objective": "Learn to dance",
        },
      ],
      "id": 4100,
      "name": "Find Success",
      "plannedEndDate": "1963-06-16",
      "plannedStartDate": "1970-04-11",
      "quarterlyFFP": Array [],
      "schedule": Array [],
      "standardsAndConditions": null,
      "statePersonnel": Array [],
      "summary": null,
    },
    Object {
      "alternatives": null,
      "contractorResources": Array [],
      "costAllocation": Array [],
      "costAllocationNarrative": Object {
        "methodology": null,
        "otherSources": null,
      },
      "description": "More gunk",
      "expenses": Array [],
      "files": Array [],
      "fundingSource": null,
      "goals": Array [],
      "id": 4101,
      "name": "My Second Activity",
      "plannedEndDate": null,
      "plannedStartDate": null,
      "quarterlyFFP": Array [],
      "schedule": Array [],
      "standardsAndConditions": null,
      "statePersonnel": Array [],
      "summary": null,
    },
  ],
  "federalCitations": null,
  "id": 4000,
  "incentivePayments": Array [],
  "keyPersonnel": Array [],
  "narrativeHIE": null,
  "narrativeHIT": null,
  "narrativeMMIS": null,
  "period": null,
  "previousActivityExpenses": Array [],
  "previousActivitySummary": null,
  "programOverview": "new overview",
  "state": "mn",
  "stateProfile": Object {
    "medicaidDirector": Object {
      "email": null,
      "name": null,
      "phone": null,
    },
    "medicaidOffice": Object {
      "address1": null,
      "address2": null,
      "city": null,
      "state": null,
      "zip": null,
    },
  },
  "status": "draft",
  "years": null,
}
`;

exports[`APD endpoint | PUT /apds/:id when authenticated as a user with permission with an APD in a state other than the user's state 1`] = `undefined`;
